packages/eslint-plugin/docs/rules/method-signature-style.md

```diff
@@ -0,0 +1,77 @@
+# Enforces using a particular method signature syntax. (`method-signature-style`)
+
+There are two ways to define an object/interface function property.
+
+```ts
+// method shorthand syntax
+interface T1 {
+  func(arg: string): number;
+}
+
+// regular property with function type
+interface T2 {
+  func: (arg: string) => number;
+}
+```
+
+A method and a function property of the same type behave differently.
+Methods are always bivariant in their argument, while function properties are contravariant in their argument under `strictFunctionTypes`. See the reasoning behind that [here](https://github.com/microsoft/TypeScript/pull/18654).
```

Comment:
Would be good to clarify the functionality with/without the flag, and better define bivariance and contravariance, as most users wouldn't know what those mean (I always forget and have to google it 😅).

----

With `strictFunctionTypes` turned off, both method signatures and function types are considered bivariant in their argument types (meaning that all arguments can be the same, supertypes, or subtypes).
However, with `strictFunctionTypes` turned on, method signatures remain bivariant, but function types become contravariant in their argument types (meaning that all arguments must be the same type, or a supertype).
See the reasoning behind that [in the typescript PR for the compiler option](https://github.com/microsoft/TypeScript/pull/18654).

This difference creates a possible safety hole when using method types instead of function types.

Reply:
I didn't want to overwhelm the user with too much information. I changed this part to be more concise and only nudge the user towards using the more correct option. I think your explanation goes too much into detail. Tell me what you think.

That original paragraph was from TypeScript's FAQ. If the user wants to learn more they still can click the link.